### DISCUSSION TITLE
GridAggregator

### DISCUSSION THREAD

**Discussion by MirenLurBarquin** [TorchIO-project/torchio#1062] [Q&A]
Hi all, 

in TorchIO [documentation](https://torchio.readthedocs.io/patches/patch_inference.html#gridsampler) there is an example of how to go back to the initial volume after using a `GridSampler` by using `tio.inference.GridAggregator`.  In this example we are just working with one subject. 

How could we go back to the initial volumes if we have a tio.data.SubjectsDataset containing multiple objects? For example after using a 'tio.Queue' and 'torch.utils.data()'. Is there any better way of doing it than with a dictionary of aggregators and creating subject_ids? 

Thank you in advance. Best :)

**Comment by fepegar** [comment]
Hi, @MirenLurBarquin. The aggregator is for inference; the queue is for training. Can you please elaborate? What are you trying to do?

**Reply by MirenLurBarquin** [reply]
Hi @fepegar, thank you for your help. 

I am doing axon segmentation. Sorry, I mixed train and inference on my first attempt. I think now I realized how should I do it: after training, if I want to do inference and visualize the results in more than one subject, I just do inference and use the aggregator for each of them one at a time, right? Following the [example](https://torchio.readthedocs.io/patches/patch_inference.html#gridsampler) in the documentation.

**Reply by fepegar** [reply]
That sounds right. Please go through the tutorials and the documentation :)
